lib/jekyll/reader.rb

```diff
@@ -131,16 +132,21 @@ def filter_entries(entries, base_directory = nil)
     # subfolder - The String representing the directory to read.
     #
     # Returns the list of entries to process
-    def get_entries(dir, subfolder)
+    def get_entries(dir, subfolder, limit = 0)
       base = site.in_source_dir(dir, subfolder)
       return [] unless File.exist?(base)
 
       entries = Dir.chdir(base) { filter_entries(Dir["**/*"], base) }
       entries.delete_if { |e| File.directory?(site.in_source_dir(base, e)) }
+
+      entries_limit = entries.length < limit ? entries.length : limit
```

Comment:
this can be rewritten has `entries_limit = [entries.length, limit].min`
`

Reply:
@shushugah The use of `Array#min` was suggested to be replaced in a previous review to avoid unnecessary allocations.